juju/testing/utils.go

```diff
@@ -84,11 +85,16 @@ func AddSubnetsWithTemplate(c *gc.C, st *state.State, numSubnets uint, infoTempl
 			return buf.String()
 		}
 
-		info.ProviderId = network.Id(permute(string(info.ProviderId)))
+		info.ProviderId = corenetwork.Id(permute(string(info.ProviderId)))
 		info.CIDR = permute(info.CIDR)
-		info.AvailabilityZone = permute(info.AvailabilityZone)
 		info.SpaceName = permute(info.SpaceName)
 
+		zones := make([]string, len(info.AvailabilityZones))
```

Comment:
The new common type unifies availability zones as a slice where the state type was singular.